src/or/hs_common.c

```diff
@@ -1617,7 +1617,7 @@ hs_pick_hsdir(smartlist_t *responsible_dirs, const char *req_key_str)
     time_t last = hs_lookup_last_hid_serv_request(dir, req_key_str, 0, 0);
     const node_t *node = node_get_by_id(dir->identity_digest);
     if (last + hs_hsdir_requery_period(options) >= now ||
-        !node || !node_has_descriptor(node)) {
+        !node || !node_has_preferred_descriptor(node, 0)) {
```

Comment:
Does this need a comment about Tor2web, similar to the last comment?

Reply:
03494da6e0be1b0208804850298b2d035a830228 is the fixup for this.